airflow/models/taskinstance.py

```diff
@@ -131,6 +133,24 @@ def clear_task_instances(tis,
 TaskInstanceKeyType = Tuple[str, str, datetime, int]
 
 
+def get_user_defined_execute_context():
+    path_to_user_context = conf.get("core", "user_defined_execute_context")
+    if path_to_user_context:
+        try:
+            module_path, attr_name = path_to_user_context.split(':')
+            module = importlib.import_module(module_path)
+            method = getattr(module, attr_name)
+            ctx_obj = method()
+            if hasattr(ctx_obj, "__enter__") and hasattr(ctx_obj, "__exit__"):
+                return ctx_obj
+            else:
+                raise AirflowException(f"Loaded function {path_to_user_context} "
+                                       f"as user context, but it does not have "
+                                       f"__enter__ or __exit__ method!")
+        except ImportError as e:
+            raise AirflowException(f"Could not import user execute context class {path_to_user_context}!", e)
```

Comment:
Can we use `airflow.utils.module_loading.import_string`?

Reply:
Also we can use this new functionality https://github.com/apache/airflow/pull/8560#discussion_r415909942

Reply:
import_string expects a "dotted path" (demands that user has path that contains at least one "."). This method captures all modules, even without ".".
Do you think it's more correct to improve import_string?
```
def import_string(required_dot=False):
...
```

Reply:
@jonathanshir in airflow we try to follow the dot pattern. As you can see the PR which contained `getimport` also unified the way of passing such values -> no `:` only `.`